File: src/eqn/ast/EqnAstNodeUnaryFunction.java
package eqn.ast;

import eqn.parser.exception.EqnException;
import org.jetbrains.annotations.NotNull;

abstract public class EqnAstNodeUnaryFunction extends EqnAstNode {

    public EqnAstNodeUnaryFunction(String value, Type type, @NotNull EqnAstNode equationNode, PrecedenceType precedenceType) throws EqnException {
        super(value, type, precedenceType);
        addOperand(equationNode);
    }

    EqnAstNode operand() {
        return getNodeAt(0);
    }
}

File: src/eqn/ast/EqnAstNodeUnaryFunction.kt
package eqn.ast

import eqn.ast.base.EqnAstNode
import eqn.ast.base.EqnAstNodeUnary

abstract class EqnAstNodeUnaryFunction(value: String, type: Type, operand: EqnAstNode, precedenceType: PrecedenceType) : EqnAstNodeUnary(value, type, precedenceType, operand)